src/ngx_http_graphite_module.c

```diff
@@ -1813,8 +1929,14 @@ ngx_http_graphite_timer_event_handler(ngx_event_t *ev) {
 
             if (nl > part) {
 
-                if (sendto(fd, part, nl - part + 1, 0, (struct sockaddr*)&sin, sizeof(sin)) == -1)
-                    ngx_log_error(NGX_LOG_ALERT, ev->log, 0, "graphite can't send udp packet");
```

Comment:
To avoid checking multiple times for the protocol type you could extract the logic to functions like `handle_tcp` and `handle_udp` and initialize a function pointer  `handler` together with the socket initialization.